core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1308,6 +1313,17 @@ class KafkaApis(val requestChannel: RequestChannel,
     }
   }
 
+  def handleInitPidRequest(request: RequestChannel.Request): Unit = {
+    val initPidRequest = request.body[InitPidRequest]
+    // Send response callback
+    def sendResponseCallback(result: InitPidResult): Unit = {
+      val responseBody: InitPidResponse = new InitPidResponse(result.error, result.pid, result.epoch)
+      trace(s"Generated new PID ${result.pid} from InitPidRequest from client ${request.header.clientId}")
+      requestChannel.sendResponse(new RequestChannel.Response(request, responseBody))
+    }
+    txnCoordinator.handleInitPid(initPidRequest.transactionalId, sendResponseCallback)
```

Comment:
I added a JIRA to track this in a future PR: https://issues.apache.org/jira/browse/KAFKA-4970